### DISCUSSION TITLE
Anti-Flickering settings are not working with v20 ?

### DISCUSSION THREAD

**Discussion by Johnny1982fd** [awawa-dev/HyperHDR#842] [Q&A]
@awawa-dev

Addon works with PI 5 and LE 12 - thank you very much!

As i wrote in previous posts, AntiFlickering Settings are not working any longer.

I made two videos with exact same settings.
The first one is HyperHDR 19, second one ist with HyperHDR 20. Same settings (imported via json)

Please have a look!

<details>

HyperHDR 19:
https://youtu.be/vwJ2x6GWqIA

HyperHDR 20:
https://youtu.be/OzpXvS2IW08

</details>

Anti Flickering settings (on both systems!)

<details>

![grafik](https://github.com/awawa-dev/HyperHDR/assets/55974883/8b640302-b0cb-4759-bcf7-35ef6df9a929)

</details>

**Comment by Johnny1982fd** [comment]
You mean increasing this value?

<details><summary>Details</summary>
<p>

![grafik](https://github.com/awawa-dev/HyperHDR/assets/55974883/726f2c61-6a4e-4bce-afa9-e6f2f1e81860)


</p>
</details> 

If yes, biggest value 8 has no change. Flickering still exists. :(

**Comment by awawa-dev** [comment]
No, backlight threshold which corresponds to the minimal backlight in the processing tab (default was 2 in v19, now it's 1 in v20, may be related to effectiveness of anti-flickering filter in lowest range). You also have some not-default settings in the smoothing tab (continues output enabled, timeout = 0, who knows what's more elsewhere). I've just tested anti-flickering filter on my v20 and it definitely works fine.

These test-only settings can confirm it'w working and can be even visible:
![obraz](https://github.com/awawa-dev/HyperHDR/assets/69086569/9c6d56e5-db84-4897-9796-b657f91ee862)
If you play some video then if you pause it, some LEDs will be updated suddenly after 2 seconds (these LEDs were frozen to eliminate "flickering" for a such configuration and were updated after a timeout, 255 means it worked for all the color range). Not for every scene, but it's rather often case for darker scenes that were changing.

**Reply by Johnny1982fd** [reply]
Ok, ill try a default installation.

can i reset to default or have i to go to deinstall and new install the addon?

**Reply by awawa-dev** [reply]
You must delete HyperHDR DB file. The path can be found at the beginning of the log.

**Reply by Johnny1982fd** [reply]
Ill give it a try.

Another question due to that: 
If i pay money, can you create a LE12 addon with HyperHDR 19?

**Reply by awawa-dev** [reply]
First try this setting because the anti-flickering filter has not changed in v20. BTW did you use exactly the same (or built-in) LUT table for v19 and v20?

**Reply by Johnny1982fd** [reply]
I have a 1080p without HDR and didnt change LUT settings in V19 or 20.

I will take some photos of my v19 Version, going to a default v20 and change settings manually (dont use the json import).

If this dont work i will pay for a LE 12 with V19 version.

**Reply by awawa-dev** [reply]
I'm here for hobby only and to help to solve issues if I can, but sometimes it's not possible due to limited details or if I cant replicate the issue: not to make money here. The minimal backlight setting was change automatically from 2 to 1 when you imported config from v19 to v20: no action was needed. So you can try to change it finally to see if it's difference because otherwise all the talk is pointless as it's the only thing beside the video processing/LUT/grabber that can affect anti-flickering filter and border detection indirectly (does it work now correctly or not): not recommend to test on the source that have black borders, one problem at the time. Or the LED driver issue? Photos are not necessary but JSON backup maybe could tell something.

**Reply by awawa-dev** [reply]
> didnt change LUT settings in V19 or 20.

that's not what I asked for

**Reply by awawa-dev** [reply]
>  Photos are not necessary but JSON backup maybe could tell something.

...and full logs taken after 2-3 minutes after the HyperHDR start. From v19 and v20 to compare them.

**Reply by Johnny1982fd** [reply]
I know, but i can compile a version by myself. I would like to test LE 12 with V19 - just to see there is a difference.
So, i know it takes time, but i only can pay for it.

Change minimal backlight has no changes in my setting.

The idea was to take phots from my version 19, install v20 and change these settings manually as the same as i have it on v19.
(No Import via json File)

Hmm, here is my Config.
[HyperHDR-19.0.0.0-Backup-2024-03-22.json](https://github.com/awawa-dev/HyperHDR/files/15205772/HyperHDR-19.0.0.0-Backup-2024-03-22.json)

**Reply by awawa-dev** [reply]
I restore it on my system. You had backlight completely disabled and by design (**in every version**) anti-flickering filter is disabled when the target from the grabber for a single LED is real black color. Which is possible only when the minimal backlight is **disabled**. I tested it only few minutes on some anime series (after setting minimal backlight to 2) and didnt notice flickering on my system that is related to some discrete changes on similar frames in series due to colors dithering/averaging error. But of course HyperHDR reacted very fast for such smoothing settings, maybe you need to increase the delay although it was the same in v19 but the LED processing was optimized and it's faster now.
I am not responsible for the WLED driver because it is not recommended by HyperHDR and it is even in the wiki, in version v20 the protocol was changed to Audio Reactive, which is in the v20 changelog, although it should not cause any changes at least WLED documentation doesnt mention anythigh about it (UDP stream remained the same only port is diffent and you should notice a change on your device WLED panel between v19 and v20 when the stream is active)... although I don't know for sure, I don't use WLED... there are a lot of settings that can affect LED rendering and I cant help here.

**Reply by awawa-dev** [reply]
BTW Gamma red/blue/green were also non-default (2.2) and extremely distorted colors... in HyperHDR we use 1.5. Neutral is 1 (grabber color = LED color).

![obraz](https://github.com/awawa-dev/HyperHDR/assets/69086569/2eef9721-61ec-413d-9196-3e0a7b558668)

more chances for the real black color (0) and disable the anti-flickering filter for a single LED when the minimal backlight is disabled...

Verify if the black border detection is still working for you.

**Comment by Johnny1982fd** [comment]
> > didnt change LUT settings in V19 or 20.
> 
> that's not what I asked for

Ok, i guess i dont understand what you mean, can you give me a little bit more infos?



> You had backlight completely disabled and by design (**in every version**) anti-flickering filter is disabled when the target from the grabber for a single LED is real black color. Which is possible only when the minimal backlight is **disabled**.

Before i missunderstand that point ... these option is meant, right?

<img width="1658" alt="Bildschirmfoto 2024-05-04 um 10 36 53" src="https://github.com/awawa-dev/HyperHDR/assets/55974883/007a8e17-e631-4b13-9fa1-4bf276aa1839">

If this is disabled, theres no Anti-Flickering filter.

**Reply by awawa-dev** [reply]
That's how it looks for your backup in **v20**, so anti-flickering filter is disabled if the color target is 0 (it still works for other values under the anti-flickering threshold)
![obraz](https://github.com/awawa-dev/HyperHDR/assets/69086569/065f1053-0210-4282-bcbb-cf3355fec8d2)

**Reply by awawa-dev** [reply]
> Ok, i guess i dont understand what you mean, can you give me a little bit more infos?

Did you stick to built-in LUT or installed a LUT using new downloader from v20.

And again: full logs for v19 & v20 from testing could be helpful and test it on the video without black border with black border in HyperHDR detection disabled, because incorrect black border detection settings (depends on the video processing ex. installed LUT, grabber settings, video format etc) could result in constant image cropping jumping causing flickering and that should be visible in the logs. So far I cannot reproduce the problem on my setup, but I dont use WLED only HyperSPI, have different grabber Elgato HD60x and dont use black border detection because the video player cuts the border and zooms the frame.

**Reply by Johnny1982fd** [reply]
I need some time to do all the tests. Pls dont close, feedback will come :)

**Reply by awawa-dev** [reply]
Ok, one more thing that could be related: try to reduce capturing rate to 30 or 20 FPS and make sure that the setting is accepted by the grabber in logs. Capturing rate also drives smoothing processing.

**Reply by Johnny1982fd** [reply]
Logs -V19:

https://pastebin.com/WrY51zZU

Logs - V20:

https://pastebin.com/cmPQUfAi

I set the settings exact as V19, but get flickering.
I have to activate Blacklight threshold.

**Reply by awawa-dev** [reply]
So you enabled Backlight threshold and it works ok now? Antiflikering filter from the start required this feature to work with max performance. Why it work in v19? Maybe because frame never reach black colour in v19, there were some minor bug correction related to led calibration in some rare cases. Anyway if you use brightness limiter (or amperage limiter) in wled and if wled scales the brightness in all the brightness scale then it invalidates anti-flickering filter **completely** and the result is unpredicted.

**Reply by Johnny1982fd** [reply]
Yes, i MUST enable backlight threshold in V20.

The reason, why i disabled it at V19 is, that black scene in movies was black in LEDs.
Now if i go to 1% i have complete low white LEDs, indenpend the scene is black or not.

**Reply by awawa-dev** [reply]
From my point of view as the developer who designed and implemented this filter something wrong is with your v19 setup: HyperHDR behaves as the minimal backlight was active or it increases a brightness (fixed bug in LED calibration?) a bit and the typical scene never reaches the black color. Of course it depends on the video processing also so eg different LUT also can cause such an effect. If led (with minimal backlight disabled: Backlight threshold =0) reaches colour (0,0,0) HyperHDR immediately power soff this led despite active anti-flickering filter and that must be visible for every user as flickering if something is starting dancing eg. between (0,0,0) and (1,1,1). On the other hand anti-flickering filter disabled such case for colours in range other than (0,0,0) so it won't happen eg for (1,1,1) and (2,2,2).

**Reply by Johnny1982fd** [reply]
Youre the chief :)

So, i have to live with that in v20
- Low white color on every LED with anti-flickering
- "Black" LEDs in black scene without anti-flickering

**Reply by awawa-dev** [reply]
The problem with the black color is that it is completely off the scale and has to be completely cut out to avoid flickering in every case, I didn't force it in the filter setting (only assume that the user wont change default backlight threshold settings  )and I see that maybe it was a mistake as it can lead to misunderstanding. Maybe I'll change that.
Coming back to this case (0,0,0), the LED is off, (1,1,1) is slightly on, but the difference is huge and it must blink, there is no such jump with (1,1,1) and (2, 2,2) because in both cases the diodes are slightly lit. Maybe when RGBW LED strips with 48-bit resolution appear en masse, this can be solved differently.

**Reply by Johnny1982fd** [reply]
Im still a little bit testing with Gamma ...

is in V20 an expert mode like V19? (didnt find it)?

**Reply by awawa-dev** [reply]
there are 3 gammas in the processing tab (reg/green/blue). try to set them to 1. And disable max brightness in WLED.

**Reply by Johnny1982fd** [reply]
HyperHDR overrite settings. Brightness is 220/255.

Question: Why do you choose 1,5 Gamma as defaut? Isnt it 2.2 ?

**Reply by awawa-dev** [reply]
> HyperHDR overrite settings. Brightness is 220/255.

you set it in the WLED driver in HyperHDR.

> Question: Why do you choose 1,5 Gamma as defaut? Isnt it 2.2 ?

Does your 2.2 come from Hyperion NG? Why they use such high value? Have no clue. Why 1.5? Because I like it more and is more neutral that 2.2 and doesn't ruin dark scenes. And the billion-dollar question: what gamma do LEDs have? Because we know it's 2.2 for monitors. And they certainly don't have the same one as LEDs.

**Reply by Johnny1982fd** [reply]
I have found some settings which are OK for me.
Discussion can be closed. 

Thanks for the good support!
